gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagProcessingEngine.java

```diff
@@ -140,6 +153,11 @@ public void run() {
           dagTask.conclude();
         } catch (Exception e) {
           log.error("DagProcEngineThread encountered exception while processing dag " + dagProc.getDagId(), e);
+          if (KafkaJobStatusMonitor.isThrowableInstanceOf(e, this.nonRetryableExceptions)) {
+            // conclude the lease so that it is not retried, if the dag proc fails with non-transient exception
```

Comment:
suggest to log (although better would be within the predicate impl (i.e. `Exception -> boolean`)